database/sql/V048__create_settlement_rewards.sql

```diff
@@ -12,20 +14,20 @@ CREATE TABLE settlement_observations (
   -- table, read from the transaction receipt
   block_number bigint NOT NULL,
   log_index bigint NOT NULL,
-  gas_used numeric(78,0) NOT NULL,
-  effective_gas_price numeric(78,0) NOT NULL,
+  gas_used numeric(78,0),
+  effective_gas_price numeric(78,0),
   -- the surplus observed from the transaction call data,
   -- and converted to ETH with the auction external prices.
-  surplus numeric(78,0) NOT NULL,
-  fee numeric(78,0) NOT NULL,
+  surplus numeric(78,0),
+  fee numeric(78,0),
 
```

Comment:
Removed `NOT NULL` because the table is initially populated (block_number and log_index) when GPv2Settlement event is emitted, while the rest is populated in an autopilot background task similar to `AuctionTransactionUpdater`.

Reply:
It was my impression that this table will not be populated when no corresponding transaction exists. This is why we had set these fields to NOT NULL. If this table is going to be populated with vacant data. I may have to change the data export queries to match this.

Reply:
It will be populated with the vacant data only initially when our system detects mined transaction.
Then, in the background task it will be populated with full data following the FINALIZED_BLOCK rule, meaning it will be lagging for 64 blocks. 
Since you are doing your queries on a historic data I think there is no need to change them.

Reply:
This isn't the right approach. See my review below.